studio/components/interfaces/Database/Functions/CreateFunction.tsx

```diff
@@ -757,6 +723,13 @@ const InputDefinition = observer(({}) => {
 const SelectSchema = observer(({}) => {
   const _localState = useContext(CreateFunctionContext)
 
+  const { project } = useProjectContext()
+  const { data } = useSchemasQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+  const schemas = data?.filter((schema) => !EXCLUDED_SCHEMAS.includes(schema.name)) ?? []
+
   return (
     <Listbox
```

Comment:
This Listbox for schemas looks like it's duplicated on a lot of places. We could add a component which fetches the schemas, renders loading state while fetching and show schemas in a listbox. Just an idea, we'll handle it in a separate PR.

Reply:
ahh yes good point actually - something SchemaSelection component